src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this.buckets = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int bucketIndex = getBucketIndex(key, buckets.length);
+        Node currentNode = null;
+        if (buckets[bucketIndex] != null) {
+            currentNode = getNodeByKey(bucketIndex, key);
+        }
+        if (currentNode != null) {
+            currentNode.value = value;
+            return;
+        }
+        currentNode = new Node<>(key, value);
+        addNode(bucketIndex, currentNode);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getBucketIndex(key, buckets.length);
+        Node currentNode = getNodeByKey(index, key);
+        if (currentNode != null) {
+            return (V) currentNode.value;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addNode(int index, Node node) {
+        if (buckets[index] != null) {
+            Node currentNode = buckets[index];
+            linkNodes(currentNode, node);
+            size++;
+            resize();
+            return;
+        }
+        buckets[index] = node;
+        size++;
+        resize();
+    }
+
+    private Node getNodeByKey(int index, K key) {
+        Node currentNode = buckets[index];
+        while (currentNode != null) {
+            if (currentNode.key == key
+                    || currentNode.key != null && currentNode.key.equals(key)) {
+                return currentNode;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private int getBucketIndex(K key, int capacity) {
+        if (key == null) {
+            return 0;
+        }
+        return key.hashCode() & (capacity - 1);
```

Comment:
```suggestion
        return key == null ? 0 : key.hashCode() & (capacity - 1);
```

It looks simpler this way